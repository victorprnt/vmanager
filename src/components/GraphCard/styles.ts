import styled from 'styled-components'

export const Wrapper = styled.div`
  width: 100%;
  padding: 1rem;

  background-color: #fff;
  border-radius: 6px;

  display: flex;
  flex-direction: column;
  align-items: center;

  .graph-text {
    color: #000;
    font-weight: 400;
    margin-bottom: 0.625rem;
    letter-spacing: 0.1rem;
  }

  .graph-title {
    font-size: 2rem;
  }

  .graph-subtitle {
    font-size: 1.5rem;
  }
`
