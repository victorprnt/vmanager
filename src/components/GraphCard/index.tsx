import * as S from './styles'

const GraphCard = () => (
  <S.Wrapper>
    <h2 className="graph-title graph-text">Graph Title</h2>
    <h3 className="graph-subtitle graph-text">subtitle</h3>
    <img src="" alt="Graph image" />
  </S.Wrapper>
)

export default GraphCard
