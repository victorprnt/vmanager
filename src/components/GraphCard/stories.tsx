// import { Story, Meta } from '@storybook/react'
import GraphCard from '.'

// export default {
//   title: 'GraphCard',
//   component: GraphCard
// } as Meta

// export const Default: Story = () => <GraphCard />
