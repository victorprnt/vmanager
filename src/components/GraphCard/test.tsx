import { render, screen } from '@testing-library/react'

import GraphCard from '.'

describe('<GraphCard />', () => {
  it('should render the heading', () => {
    const { container } = render(<GraphCard />)

    expect(
      screen.getByRole('heading', { name: /Graph title/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /subtitle/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('img', { name: /graph image/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
