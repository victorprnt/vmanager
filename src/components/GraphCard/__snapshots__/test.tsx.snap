// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GraphCard /> should render the heading 1`] = `
<div
  class="sc-bczRLJ iZVTKG"
>
  <h2
    class="graph-title graph-text"
  >
    Graph Title
  </h2>
  <h3
    class="graph-subtitle graph-text"
  >
    subtitle
  </h3>
  <img
    alt="Graph image"
    src=""
  />
</div>
`;
